src/commonMain/kotlin/com/charleskorn/kaml/YamlConfiguration.kt

```diff
@@ -83,12 +85,35 @@ public enum class SingleLineStringStyle {
     DoubleQuoted,
     SingleQuoted,
     Plain,
-    ;
+
+    /**
+     * This is the same as [SingleLineStringStyle.Plain], except strings that are not strictly strings in YAML
+     * without quotes will be escaped with the escape style defined in [AmbiguousEscapeStyle].
+     *
+     * True -> Boolean -> Escaped
+     *
+     * 0xAB -> Hexadecimal -> Escaped
+     *
+     * 1 -> Int -> Escaped
+     *
+     * 1.2 -> Float -> Escaped
+     *
+     * 1.2.3 -> Unambiguously a string -> Not escaped
+     *
+     * @since 0.51.0
+     */
+    PlainExceptAmbiguous;
 
     public val multiLineStringStyle: MultiLineStringStyle
         get() = when (this) {
             DoubleQuoted -> MultiLineStringStyle.DoubleQuoted
             SingleQuoted -> MultiLineStringStyle.SingleQuoted
             Plain -> MultiLineStringStyle.Plain
+            PlainExceptAmbiguous -> MultiLineStringStyle.Plain
         }
 }
+
+public enum class AmbiguousEscapeStyle {
```

Comment:
Sorry, just had another thought on the name of this - how about `AmbiguousQuoteStyle`?